### ISSUE TITLE
[BUG][JAVA] Nested anyOf object cannot be created

### ISSUE THREAD

**Issue by shaguar** [OpenAPITools/openapi-generator#17070]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [ ] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description
Nested anyOf does not create a proper POJO model . The generated POJO source refered to a AnyOfExternalTextlinkProductsTextlinkInternalTextlinkHeadline.java class which is not generated. Is there a issue with the generation of nested anyOf items?

Do I miss a configuration parameter which is neede for this kind of nested objects?

##### openapi-generator version

7.0.1

##### OpenAPI declaration file content or url
```yaml
components:
  schemas:
    Page:
      allOf:
        - $ref: '#/components/schemas/Details'
        - type: object
          properties:
            linkListColumn1:
              type: object
              properties:
                value:
                  type: array
                  items:
                    anyOf:
                      - $ref: '#/components/schemas/ExternalTextlink'
                      - $ref: '#/components/schemas/ProductsTextlink'
                      - $ref: '#/components/schemas/InternalTextlink'
                      - $ref: '#/components/schemas/Headline'
              required:
                - value
```

##### Generation Details
```
java -jar openapi-generator-cli.jar generate -g java -i openapi.yaml -c config.yaml --library resttemplate -o gensrc/ --global-property apis,models,apiDocs=false,apiTests=false,modelDocs=false,modelTests=false
```

**Comment by wing328** [comment]
can you please share the full spec to more easily reproduce the issue?

**Comment by shaguar** [comment]
Hi, below you will find the full spec, thank you 
```yaml
openapi: 3.0.3
info:
  title: My Title
  version: 1.0.0
servers:
  - url: https://localhost
security:
  - basicAuth: []
paths:
  /{index}/_search:
    post:
      operationId: Search_Post_WithIndex
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/BodyParams'
      parameters:
        - name: index
          in: path
          schema:
            type: string
            enum:
              - indexA
              - indexB
              - indexC
          required: true
      responses:
        '200':
          description: Search_Post_WithIndex 200 response
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Search_Post'
        '400':
          description: bad input parameter
components:
  schemas:
    PageGcFooterSettingsMaster:
      allOf:
        - $ref: '#/components/schemas/DefaultElementDetails'
        - type: object
          properties:
            linkListColumn1:
              type: object
              properties:
                value:
                  type: array
                  items:
                    anyOf:
                      - $ref: '#/components/schemas/ExternalTextlink'
                      - $ref: '#/components/schemas/ProductsTextlink'
                      - $ref: '#/components/schemas/InternalTextlink'
                      - $ref: '#/components/schemas/Headline'
              required:
                - value
    ExternalTextlink:
      allOf:
        - $ref: '#/components/schemas/DefaultLinkTemplate'
        - type: object
          properties:
            config:
              type: object
            reference:
              $ref: '#/components/schemas/Text'
            text:
              $ref: '#/components/schemas/Text'
            target:
              $ref: '#/components/schemas/Option'
    ProductsTextlink:
      allOf:
        - $ref: '#/components/schemas/DefaultLinkTemplate'
        - type: object
          properties:
            config:
              type: object
            linkType:
              $ref: "#/components/schemas/Toggle"
            text:
              $ref: "#/components/schemas/Text"
    InternalTextlink:
      allOf:
        - $ref: '#/components/schemas/DefaultLinkTemplate'
        - type: object
          properties:
            config:
              type: object
            linkType:
              $ref: "#/components/schemas/Option"
            reference:
              $ref: "#/components/schemas/Reference"
            text:
              $ref: "#/components/schemas/Text"
    Headline:
      allOf:
        - $ref: '#/components/schemas/DefaultLinkTemplate'
        - type: object
          properties:
            config:
              type: object
            text:
              $ref: "#/components/schemas/Text"
    DefaultLinkTemplate:
      type: object
      properties:
        type:
          type: string
      required:
        - type
    DefaultElementDetails:
      type: object
      properties:
        contentId:
          type: string
        id:
          type: string
        type:
          type: string
        config:
          type: object
      required:
        - contentId
        - id
        - language
        - country
        - baseType
        - revision
        - type
    Text:
      type: object
      properties:
        content:
          type: string
        type:
          type: string
      required:
        - type
    Option:
      type: object
      properties:
        content:
          type: string
        type:
          type: string
      required:
        - type
    Toggle:
      type: object
      properties:
        content:
          type: boolean
        type:
          type: string
      required:
        - type
    Reference:
      type: object
      properties:
        type:
          type: string
        projectId:
          type: number
      required:
        - type
        - projectId
    Search_Post:
      type: object
      properties:
        took:
          type: integer
          format: int64
        timed_out:
          type: boolean
        hits:
          $ref: '#/components/schemas/HitsMetadata'
    HitsMetadata:
      type: object
      properties:
        total:
          $ref: '#/components/schemas/Total'
        hits:
          type: array
          items:
            $ref: '#/components/schemas/Hits'
    Hits:
      type: object
      properties:
        _index:
          type: string
        _type:
          type: string
        _id:
          type: string
        _source: {}
        fields: {}
    Total:
      type: object
      properties:
        value:
          type: integer
          format: int32
        relation:
          $ref: '#/components/schemas/Relation'
    Relation:
      type: string
      enum:
        - eq
        - gte
    Time:
      type: string
      enum:
        - d
        - h
        - m
        - s
        - ms
        - micros
        - nanos
    BodyParams:
      type: object
      properties:
        explain:
          type: boolean
        from:
          type: integer
          format: int32
        size:
          type: integer
          format: int32
        timeout:
          $ref: '#/components/schemas/Time'
        version:
          type: boolean
        fields:
          type: array
          items:
            type: string
        query:
          $ref: '#/components/schemas/UserDefinedStructure'
    UserDefinedStructure:
      type: object
      properties:
        match:
          $ref: '#/components/schemas/UserDefinedMap'
    UserDefinedMap:
      type: object
      additionalProperties: {}
  securitySchemes:
    basicAuth:
      type: http
      scheme: basic
```

**Comment by wing328** [comment]
fixed by #17138

please give it another try with the latest master

**Comment by shaguar** [comment]
@wing328 it worked with the latest master. Thank you
